src/compiler/compile/Component.ts

```diff
@@ -1125,12 +1137,14 @@ export default class Component {
 		for (const [name, node] of top_level_function_declarations) {
 			if (is_hoistable(node)) {
 				const variable = this.var_lookup.get(name);
-				variable.hoistable = true;
-				hoistable_nodes.add(node);
+				if (!('argument' in variable && variable.argument)) {
```

Comment:
There's no need for the `'argument' in variable` part, but what about moving this to the outer `if` or inside `is_hoistable`?

Reply:
Done.